sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -161,6 +164,46 @@ case class GlobalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
   override def outputOrdering: Seq[SortOrder] = child.outputOrdering
 }
 
+/**
+ * Skip the first `offset` elements then take the first `limit` of the following elements in
+ *  the child's single output partition.
+ */
+case class GlobalLimitAndOffsetExec(
+    limit: Int,
+    offset: Int,
+    child: SparkPlan) extends BaseLimitExec {
+
+  override def requiredChildDistribution: List[Distribution] = AllTuples :: Nil
+
+  override def outputPartitioning: Partitioning = child.outputPartitioning
+
+  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
+
+  override def doExecute(): RDD[InternalRow] = {
+    val rdd = child.execute().mapPartitions { iter => iter.take(limit + offset)}
+    val skips = rdd.take(offset)
```

Comment:
To implement this feature, we shouldn't probably keep such records within the memory. For example, if the offset was a huge number with 1 as limit, it will throw OOM.

Reply:
Yes, but if the limit was a huge number, it still will throw OOM.
This PR only follows the implement of `BaseLimitExec` https://github.com/apache/spark/blob/14ca8d326fcc5035da5b7a99772aacff8edad163/sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala#L109
If this is an issue, `GlobalLimitExec` and `LocalLimitExec` need to resolve too.